### PR TITLE
Relay indexer pubsub messages 

### PR DESCRIPTION
**Pull request by gammazero**
This uses /indexer/ingest/<networkname> for pubsub topic to subscribe to. Test publishers should use a differnt network namd to distinguish them from 'real' publishers.

This replaces PR #7739

### CODE REVIEW

**Review by vyzo — APPROVED** [review]
this is totally fine, but dont you want the  bootstrappers to subscribe as well?

**Comment by vyzo** [comment]
Meaning do an actual subscribe so that they can do PX and fullfill their bootstrapping duties.

**Comment by gammazero** [comment]
@vyzo This side is only publishing on the ingest topic, does it need to subscribe anyway?

**Comment by gammazero** [comment]
Lotus is only every publishing to the ingest topic.  The documentation says:
> To publish a message to some topic, use Publish; you don't need to be subscribed to the topic in order to publish.
But, as a bootstrapping node a subscription to the topic is needed, otherwise they will not pass on publications to other peers?

**Comment by vyzo** [comment]
Publishing does not require a subscription indeed, and this is all fine as is.

Having said that, it would be very helpful for the bootsrappers to actually subscribe so that clients can discover peers through PX.
But this is non normative, really.

File: node/modules/services.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ func RunHello(mctx helpers.MetricsCtx, lc fx.Lifecycle, h host.Host, svc *hello.
 
 	sub, err := h.EventBus().Subscribe(new(event.EvtPeerIdentificationCompleted), eventbus.BufSize(1024))
 	if err != nil {
-		return xerrors.Errorf("failed to subscribe to event bus: %w", err)
+		return fmt.Errorf("failed to subscribe to event bus: %w", err)
```
Comment (vyzo) [inline node/modules/services.go:61]: please dont replace this.  we use xerrors because it has stack traces.

File: node/modules/services.go [hunk 2]
```diff
@@ -198,6 +198,29 @@ func HandleIncomingMessages(mctx helpers.MetricsCtx, lc fx.Lifecycle, ps *pubsub
 	waitForSync(stmgr, pubsubMsgsSyncEpochs, subscribe)
 }
 
+func RelayIndexerMessages(lc fx.Lifecycle, ps *pubsub.PubSub, nn dtypes.NetworkName) error {
+	topicName := build.IngestTopic(nn)
+	topicHandle, err := ps.Join(topicName)
+	if err != nil {
+		return fmt.Errorf("failed to join pubsub topic %s: %w", topicName, err)
```
Comment (vyzo) [inline node/modules/services.go:205]: use xerrors.

**Review by vyzo — CHANGES_REQUESTED** [review]
you need to register a validator, otherwise it will be spamfest.

File: node/modules/services.go (hunk 1 above)
Comment (gammazero) [reply node/modules/services.go]: Done

File: node/modules/services.go (hunk 2 above)
Comment (gammazero) [reply node/modules/services.go]: Done

**Comment by gammazero** [comment]
I added a validator, but am not sure if it is adequate.  I am not sure is there is anything specific the validator is expected to do, like rate limiting.

**Comment by gammazero** [comment]
Merged into PR #7313 as PR #8026

**Comment by vyzo** [comment]
@gammazero That doesn't change the fact that you need to add a vlidator! The network can be spammed by flooding junk messages to the bootstrappers on the indexer topic, which would then be gossiped to the other nodes in the indexer overlay.

**Pull request closed without merging**